### PR TITLE
[FIX][10.0]Make a defensive check before processing table and model rename

### PR DESCRIPTION
**Pull request by flotho**
closes https://github.com/OCA/sale-workflow/issues/789

### CODE REVIEW

**Review by moylop260 — APPROVED** [review]

File: base_exception/migrations/10.0.2.0.0/pre-migration.py [hunk 1]
```diff
@@ -8,5 +8,11 @@
 @openupgrade.migrate(use_env=True)
 def migrate(env, version):
     cr = env.cr
-    openupgrade.rename_tables(cr, [('sale_exception', 'exception_rule')])
-    openupgrade.rename_models(cr, [('sale.exception', 'exception.rule')])
+    
+    cr.execute("SELECT 1 FROM pg_class WHERE relname = 'sale_exception'")
```
Comment (pedrobaeza) [inline base_exception/migrations/10.0.2.0.0/pre-migration.py]: Use `openupgrade.table_exists('sale_exception')` for shorter version

File: base_exception/migrations/10.0.2.0.0/pre-migration.py [hunk 2]
```diff
@@ -8,5 +8,11 @@
 @openupgrade.migrate(use_env=True)
 def migrate(env, version):
     cr = env.cr
-    openupgrade.rename_tables(cr, [('sale_exception', 'exception_rule')])
-    openupgrade.rename_models(cr, [('sale.exception', 'exception.rule')])
+
```
Comment (moylop260) [inline base_exception/migrations/10.0.2.0.0/pre-migration.py]: Remove the space ` ` in order to fix
`base_exception/migrations/10.0.2.0.0/pre-migration.py:11:1: W293 blank line contains whitespace`

Same case for
`base_exception/migrations/10.0.2.0.0/pre-migration.py:15:1: W293 blank line contains whitespace`

please

**Comment by flotho** [comment]
Hi @pedrobaeza ,

All is green now

File: base_exception/migrations/10.0.2.0.0/pre-migration.py [hunk 3]
```diff
@@ -8,5 +8,10 @@
 @openupgrade.migrate(use_env=True)
 def migrate(env, version):
     cr = env.cr
-    openupgrade.rename_tables(cr, [('sale_exception', 'exception_rule')])
-    openupgrade.rename_models(cr, [('sale.exception', 'exception.rule')])
+    cr.execute("SELECT 1 FROM pg_class WHERE relname = 'sale_exception'")
```
Comment (pedrobaeza) [inline base_exception/migrations/10.0.2.0.0/pre-migration.py]: Remove this line as not needed anymore

File: base_exception/migrations/10.0.2.0.0/pre-migration.py [hunk 4]
```diff
@@ -8,5 +8,10 @@
...
+    if openupgrade.table_exists('sale_exception'):
+        openupgrade.rename_tables(cr, [('sale_exception', 'exception_rule')])
+
+    except_model = env['ir.model'].search([('name', '=', 'sale.exception')])
```
Comment (pedrobaeza) [inline base_exception/migrations/10.0.2.0.0/pre-migration.py]: No need of extra check. If the table exists, the model existed, so we have to rename it. Put all under the same `if openupgrade.table_exists...`.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by flotho** [comment]
Great thanks, I'm very proud of your approval.

**Comment by flotho** [comment]
Hi, @pedrobaeza @moylop260 ,

From my point of view, the travis failure are not concerning my job, am I right?

Regards

File: base_exception/migrations/10.0.2.0.0/pre-migration.py (hunk 3 above)
Comment (pedrobaeza) [reply base_exception/migrations/10.0.2.0.0/pre-migration.py]: This line should still be removed

**Comment by pedrobaeza** [comment]
Travis error is not related. @moylop260 any clue about it?

**Comment by flotho** [comment]
Thanks @pedrobaeza ,

Any chance to be merge anyway?

**Comment by pedrobaeza** [comment]
Please remove the line I have told about and I'll merge.

File: base_exception/migrations/10.0.2.0.0/pre-migration.py (hunk 3 above)
Comment (flotho) [reply base_exception/migrations/10.0.2.0.0/pre-migration.py]: Done, 
Thanks I didin't have my eyes in their place ;-)

**Review by pedrobaeza — APPROVED** [review]
And Travis meanwhile has been fixed!

**Comment by flotho** [comment]
(y)

**Pull request merged**